bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlItem.java

```diff
@@ -686,4 +688,12 @@ public static XmlItem createXMLItemWithDefaultNSAttribute(QName name, boolean re
         }
         return item;
     }
+
+    public Type getIteratorNextReturnType() {
```

Comment:
I think it's possible to move this method to `XmlValue` class and override only for sequence.